File: Diagrammatize.Design.Pattern/code/python/FactoryMethod/Sample/idcard/IDCard.py
from framework.Product import Product

class IDCard(Product):
    def __init__(self, owner):
        print("制作" + owner + "的ID卡。")
        self.owner = owner

    def use(self):
        print("使用" + self.owner + "的ID卡。")

    def getOwner(self):
        return self.owner

File: Diagrammatize.Design.Patterns/code/cxx/FactoryMethod/Sample/idcard/IDCard.cpp
#include "IDCard.hpp"
#include <iostream>

IDCard::IDCard(const std::string& owner_): owner(owner_) {
    std::cout << "制作" << owner << "的ID卡。" << std::endl;
}

void IDCard::use() {
    std::cout << "使用" << owner << "的ID卡。" << std::endl;
}

std::string IDCard::getOwner() {
    return owner;
}
